src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1583,9 +1591,12 @@ static int client_parse_message(
 
         switch (client->state) {
         case DHCP_STATE_SELECTING:
-                if (r != DHCP_OFFER)
+                if (r == DHCP_ACK && (client->anonymize || !client->rapid_commit || !lease->rapid_commit))
```

Comment:
I'd like to make the error messages more descriptive, and also check the offer message does not contain the Rapid Commit option. Something like the following.
```c
if (r == DHCP_ACK) {
        if (client->anonymize || !client->rapid_commit)
                return log_dhcp_client_errno(client, SYNTHETIC_ERRNO(ENOMSG),
                                             "received unexpected ACK, ignoring.");
        if (!lease->rapid_commit)
                return log_dhcp_client_errno(client, SYNTHETIC_ERRNO(ENOMSG),
                                             "received rapid ACK without Rapid Commit option, ignoring.");
} else if (r == DHCP_OFFER) {
        if (lease->rapid_commit)
                return log_dhcp_client_errno(client, SYNTHETIC_ERRNO(ENOMSG),
                                             "received OFFER with Rapid Commit option, ignoring");
} else
        return log_dhcp_client_errno(client, SYNTHETIC_ERRNO(ENOMSG),
                                     "received unexpected message, ignoring.");
```